subprojects/core/src/main/java/org/gradle/internal/service/scopes/WorkerSharedGlobalScopeServices.java

```diff
@@ -154,4 +161,28 @@ CurrentGradleInstallation createCurrentGradleInstallation() {
     ClassLoaderFactory createClassLoaderFactory() {
         return new DefaultClassLoaderFactory();
     }
+
+    BuildOperationListenerManager createBuildOperationListenerManager() {
+        return new DefaultBuildOperationListenerManager();
+    }
+
+    CurrentBuildOperationRef createCurrentBuildOperationRef() {
+        return CurrentBuildOperationRef.instance();
+    }
+
+    BuildOperationRunner createBuildOperationRunner(
+        Clock clock,
+        CurrentBuildOperationRef currentBuildOperationRef,
+        ProgressLoggerFactory progressLoggerFactory,
+        BuildOperationIdFactory buildOperationIdFactory,
+        BuildOperationListenerManager buildOperationListenerManager
+    ) {
+        BuildOperationListener listener = buildOperationListenerManager.getBroadcaster();
+        return new DefaultBuildOperationRunner(
+            currentBuildOperationRef,
+            clock::getCurrentTime,
+            buildOperationIdFactory,
+            () -> new BuildOperationProgressEventListenerAdapter(listener, progressLoggerFactory, clock)
+        );
+    }
```

Comment:
💭 This is the new service addition. We did create an instance of this before, too, but it was not an injectable service; instead, `DefaultBuildOperationExecutor` created it in its constructor.